### ISSUE TITLE
Keeping model matrix structure after import/export of .QB format

### ISSUE THREAD

**Issue by YarlBoro** [guillaumechereau/goxel#43]
I hope that at some point in the future Goxel can import each Qubicle matrix as separate layer and the other way around, export each layer to separate matrix while saving to QB format.

Thanks in advance!

**Comment by darkhog** [comment]
This. This is extremely important in modding certain games such as Staxel. In some ways Goxel is already superior to Qubicle, but this is required as layers/matrices have meaning for Staxel so they need to be preserved.

**Comment by guillaumechereau** [comment]
Agreed.  This is important.  I think I will make it so that the qubicle matrices are imported as separated layers, and saved the same way.

**Comment by guillaumechereau** [comment]
OK I just pushed some changes to qubicle import/export, so that now we keep the matrices as layers.
I haven't tested much yet.

**Comment by YarlBoro** [comment]
I've done some tests with files that have 12-18 matrices. Loading and saving .qb files keeping layers/matrixes works! :) 
Colors, structure and orientation seem to be fine so far.
I will report if I will encounter any bugs.

Thank You, this one is a huge time saver!

**Comment by darkhog** [comment]
How about size of the bounding box in case of matrices that have oversized bounding boxes with lots of empty space around? In case of certain items for Staxel (clothes/accessories), it matters for the size of the bounding boxes to be preserved even if actual voxel thing is just few voxels in size.

**Comment by guillaumechereau** [comment]
Yes that should be taken into account in the import/export, I'll add this.
Actually, I am almost considering forcing all layers to have a bounding box, even a very large one, to avoid confusion.

**Comment by darkhog** [comment]
Yes, that would be a good thing. I mean, Qubicle does that for matrices and no one is complaining. BTW. Does Goxel compress exported QBs? Some voxel programs I know of don't do that so I want to know.

**Comment by guillaumechereau** [comment]
For the moment goxel doesn't use compression in the QB export.  Would be nice to have this as an option.
